recipes/aws-sdk-cpp/all/conanfile.py

```diff
@@ -0,0 +1,133 @@
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_folder = "aws-sdk-cpp-{}".format(self.version)
+        os.rename(extracted_folder, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+
+        if not self.settings.compiler.cppstd:
+            self._cmake.definitions["CMAKE_CXX_STANDARD"] = 11
+        self._cmake.definitions["BUILD_DEPS"] = False
+        self._cmake.definitions["ENABLE_UNITY_BUILD"] = True
+        self._cmake.definitions["ENABLE_RTTI"] = self.options.with_rtti
+        self._cmake.definitions["FORCE_CURL"] = self.options.with_curl
+        self._cmake.definitions["ENABLE_CURL_LOGGING"] = self.options.get_safe("enable_curl_logging")
+        if self.settings.compiler.cppstd:
+            self._cmake.definitions["CPP_STANDARD"] = self._cmake_cpp_standard
+
+        self._cmake.definitions["ENABLE_TESTING"] = False
+        self._cmake.definitions["SIMPLE_INSTALL"] = True
+        if self.settings.compiler == "Visual Studio":
+            self._cmake.definitions["FORCE_SHARED_CRT"] = "MD" in str(self.settings.compiler.runtime)
+
+        self._cmake.definitions["AUTORUN_UNIT_TESTS"] = False
+
+        self._cmake.definitions["ANDROID_BUILD_CURL"] = False
+        self._cmake.definitions["ANDROID_BUILD_OPENSSL"] = False
+        self._cmake.definitions["ANDROID_BUILD_ZLIB"] = False
+
+        self._cmake.configure()
+        return self._cmake
+
+    def _patch_sources(self):
+        tools.replace_in_file(os.path.join(self._source_subfolder, "cmake", "sdks.cmake"),
+                              "sort_links(EXPORTS)", "# sort_links(EXPORTS)")
+        for root, _, _ in os.walk(self._source_subfolder):
+            cmakelists = os.path.join(root, "CMakeLists.txt")
+            repl = "AWS::aws-c-event-stream"
+            if os.path.isfile(cmakelists) and repl in tools.load(cmakelists):
+                tools.replace_in_file(cmakelists, repl, "CONAN_PKG::aws-c-event-stream", strict=False)
+
+    @property
+    def _cpu_count(self):
+        try:
+            cpus = int(tools.cpu_count())
+            return min((cpus, 2))
+        except ValueError:
+            return 2
+
+    def build(self):
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        # Reduce parallelism to avoid out-of-memory on limited memory build systems
+        with tools.environment_append({"CONAN_CPU_COUNT": str(self._cpu_count)}):
+            cmake.build()
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
```

Comment:
as there are inter-dependencies between components (e.g. many depend on `aws-sdk-core`) it's not good to use `collect_libs` (see https://docs.conan.io/en/latest/reference/tools.html?highlight=inter%20dependent#tools-collect-libs)

Reply:
Maybe it needs a similar approach as my boost components pr (https://github.com/conan-io/conan-center-index/pull/2097)

Reply:
wow, you have a lot going on